overlord/ifacestate/apparmorprompting/common/common.go

```diff
@@ -0,0 +1,328 @@
+var ErrUnrecognizedFilePermission = errors.New("file permissions mask contains unrecognized permission")
+
+type OutcomeType string
+
+const (
+	OutcomeUnset OutcomeType = ""
+	OutcomeAllow OutcomeType = "allow"
+	OutcomeDeny  OutcomeType = "deny"
+)
+
+type LifespanType string
+
+const (
+	LifespanUnset    LifespanType = ""
+	LifespanForever  LifespanType = "forever"
+	LifespanSession  LifespanType = "session"
+	LifespanSingle   LifespanType = "single"
+	LifespanTimespan LifespanType = "timespan"
+)
+
+type PermissionType string
+
+const (
+	PermissionExecute             PermissionType = "execute"
+	PermissionWrite               PermissionType = "write"
+	PermissionRead                PermissionType = "read"
+	PermissionAppend              PermissionType = "append"
+	PermissionCreate              PermissionType = "create"
+	PermissionDelete              PermissionType = "delete"
+	PermissionOpen                PermissionType = "open"
+	PermissionRename              PermissionType = "rename"
+	PermissionSetAttr             PermissionType = "set-attr"
+	PermissionGetAttr             PermissionType = "get-attr"
+	PermissionSetCred             PermissionType = "set-cred"
+	PermissionGetCred             PermissionType = "get-cred"
+	PermissionChangeMode          PermissionType = "change-mode"
+	PermissionChangeOwner         PermissionType = "change-owner"
+	PermissionChangeGroup         PermissionType = "change-group"
+	PermissionLock                PermissionType = "lock"
+	PermissionExecuteMap          PermissionType = "execute-map"
+	PermissionLink                PermissionType = "link"
+	PermissionChangeProfile       PermissionType = "change-profile"
+	PermissionChangeProfileOnExec PermissionType = "change-profile-on-exec"
+)
+
+// Removes the given permission from the given list of permissions.
+// Returns a new list with all instances of the given permission removed.
+// If the given permission is not found in the list, returns an error, along
+// with the original list.
+func RemovePermissionFromList(list []PermissionType, permission PermissionType) ([]PermissionType, error) {
+	newList := make([]PermissionType, 0, len(list)-1)
+	found := false
+	for _, perm := range list {
+		if perm == permission {
+			found = true
+			continue
+		}
+		newList = append(newList, perm)
+	}
+	if !found {
+		return list, ErrPermissionNotInList
+	}
+	return newList, nil
+}
+
+// Converts the given timestamp string to a time.Time.
+// The timestamp string is expected to be of the format time.RFC3999Nano.
+// If it cannot be parsed as such, returns an error.
+func TimestampToTime(timestamp string) (time.Time, error) {
+	return time.Parse(time.RFC3339Nano, timestamp)
+}
+
+// Returns the current time as a string in time.RFC3999Nano format.
+func CurrentTimestamp() string {
+	return time.Now().Format(time.RFC3339Nano)
+}
+
+// Returns a new unique ID.
+// The ID is the current unix time in nanoseconds encoded as base32.
+func NewId() string {
```

Comment:
Should there be a custom type created for id, since all IDs should now be a `UnixNano` creation time encoded in base32?